extensions/oidc/deployment/src/test/java/io/quarkus/oidc/test/UserInfoRequiredDetectionTest.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.oidc.test;
+
+import java.util.List;
+
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.oidc.UserInfo;
+import io.quarkus.oidc.runtime.OidcConfig;
+import io.quarkus.security.PermissionsAllowed;
+import io.quarkus.test.QuarkusDevModeTest;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.keycloak.client.KeycloakTestClient;
+import io.quarkus.test.keycloak.server.KeycloakTestResourceLifecycleManager;
+import io.restassured.RestAssured;
+import io.vertx.ext.web.Router;
+
+@QuarkusTestResource(KeycloakTestResourceLifecycleManager.class)
+public class UserInfoRequiredDetectionTest {
+
+    @RegisterExtension
+    static final QuarkusDevModeTest test = new QuarkusDevModeTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClasses(UserInfoResource.class, UserInfoEndpoint.class)
+                    .addAsResource(
+                            new StringAsset(
+                                    """
+                                            quarkus.oidc.tenant-paths=/user-info/default-tenant
+                                            quarkus.oidc.user-info-path=http://${quarkus.http.host}:${quarkus.http.port}/user-info-endpoint
+                                            quarkus.oidc.named.authentication.user-info-required=true
```

Comment:
Sorry, I was surprised it worked for named tenants, but I didn't mention I left there this property from time where I wrote the test that should pass when proper solution is in place. I'll provide tweak for solution. It's easy to fix, just pity I didn't mention this previously I re-read this.